client/authority-discovery/src/worker.rs

```diff
@@ -730,22 +764,29 @@ impl Metrics {
 				)?,
 				registry,
 			)?,
-			amount_last_published: register(
+			amount_addresses_last_published: register(
 				Gauge::new(
 					"authority_discovery_amount_external_addresses_last_published",
 					"Number of external addresses published when authority discovery last \
 					 published addresses."
 				)?,
 				registry,
 			)?,
-			request: register(
+			requests: register(
 				Counter::new(
 					"authority_discovery_authority_addresses_requested_total",
 					"Number of times authority discovery has requested external addresses of a \
 					 single authority."
 				)?,
 				registry,
 			)?,
+			requests_pending: register(
```

Comment:
Isn't it better to split this between two counters `requests_started_total` and `requests_finished_total`?

Reply:
_I am not sure I understand your question correctly. Let me try._

With this pull request authority discovery publishes (among others) the following 3 metrics:

- `authority_discovery_authority_address_requests_pending`

- `authority_discovery_authority_addresses_requested_total`

- `authority_discovery_dht_event_received`

Your suggested `requests_started_total` metric already exists with `authority_discovery_authority_addresses_requested_total`. Your suggested `requests_finished_total` already exists with `sum(authority_discovery_dht_event_received{name=~"value_found|value_not_found") by (name)`.

We could split `authority_discovery_authority_address_requests_pending` into two metrics in order to not miss spikes in between Prometheus scrapes, but I doubt that is worth the complexity tracking the two causes.

Does that make sense @tomaka?

Reply:
If `authority_discovery_authority_address_requests_pending` is always equal to `authority_discovery_authority_addresses_requested_total - sum(authority_discovery_dht_event_received{name=~"value_found|value_not_found") by (name)`, then what is the reason for having it?

Reply:
Ah, now I get the confusion.

There is a difference between pending, in-flight and finished lookups.

Every 10 minutes the authority discovery module requests the current and next authority set from the runtime and overrides `pending_lookups` accordingly. It then takes `MAX_IN_FLIGHT_LOOKUPS` lookups from `pending_lookups` calls the `NetworkService::get_record` and tracks those in-flight lookups in `in_flight_lookups`. Whenever a lookup succeeds another lookup is popped off from `pending_lookups` and passed to the network and inserted into `in_flight_lookups`.

`authority_discovery_authority_address_requests_pending` tracks the amount of pending, not yet started lookups.

`..._authority_addresses_requested_total - sum(..._dht_event_received{name=~"value_found|value_not_found") by (name)` tracks the amount of in-flight, started, not yet finished lookups.

Does that make sense @tomaka?